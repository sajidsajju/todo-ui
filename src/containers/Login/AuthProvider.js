import React, { useState, useEffect, createContext } from "react";
import { auth } from "../../util/firebase";

export const AuthContext = createContext();

export const AuthProvider = ({ children }) => {
  const [user, setUser] = useState(null);

  useEffect(() => {
    auth.onAuthStateChanged(setUser);
  }, []);

  return (
    <AuthContext.Provider value={{ user }}>{children}</AuthContext.Provider>
  );
};
