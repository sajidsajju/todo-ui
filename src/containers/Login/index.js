import React, { useEffect, useState } from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Button } from "@material-ui/core";

import { auth } from "../../util/firebase";
import todoist from "./todoist.jpg";
import MailOutlineIcon from "@material-ui/icons/MailOutline";
import { useHistory } from "react-router";
import { user } from "../../components/Api";

const useStyles = makeStyles({
  container: {
    width: "40vw",
    height: "55vh",
    background: "#FFFAFA",
    position: "absolute",
    left: "50%",
    top: "50%",
    transform: "translate(-50%, -50%)",
    textAlign: "center",
    borderRadius: "25px",
  },
  image: {
    width: "90%",
    height: "85%",
    marginTop: "-50px",
    borderRadius: "25px",
    marginBottom: "7%",
  },
});

const Login = () => {
  const classes = useStyles();
  const history = useHistory();

  const [logged, setLogged] = useState(false);

  useEffect(() => {
    if (logged) {
      history.push("/");
    }
  }, [history, logged]);

  auth.onAuthStateChanged(function (user) {
    if (user) {
      setLogged(true);
    } else {
      setLogged(false);
    }
  });

  return (
    <div className={classes.container}>
      <img className={classes.image} src={todoist} alt="img" />
      <Button
        style={{ backgroundColor: "#696969", color: "#FFFFFF" }}
        variant="contained"
        onClick={user}
        startIcon={<MailOutlineIcon />}
      >
        Login with Google
      </Button>
    </div>
  );
};

export default Login;
