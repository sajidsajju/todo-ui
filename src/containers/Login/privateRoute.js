import React, { useContext } from "react";
import { Redirect } from "react-router-dom";
import { AuthContext } from "./AuthProvider";

const PrivateRoute = ({ component: RouteComponent, ...props }) => {
  const { user } = useContext(AuthContext);
  const userDetails = user;
  return userDetails ? (
    <RouteComponent {...props} />
  ) : (
    <Redirect to={"/login"} />
  );
};

export default PrivateRoute;
