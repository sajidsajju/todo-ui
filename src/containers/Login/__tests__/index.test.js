import React from "react";
import { render, screen } from "@testing-library/react";

import Login from "../";

describe("<Login  />", () => {
  it("should match the snapshot", () => {
    const rendered = render(<Login />);

    expect(rendered.container.firstChild).toMatchSnapshot();
  });

  it("should match the Login Button", () => {
    render(<Login />);

    expect(screen.getByText("Login with Google")).toBeInTheDocument();
  });
});
