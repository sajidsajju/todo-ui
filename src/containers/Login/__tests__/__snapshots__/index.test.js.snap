// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login  /> should match the snapshot 1`] = `
<div
  class="makeStyles-container-1"
>
  <img
    alt="img"
    class="makeStyles-image-2"
    src="todoist.jpg"
  />
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained"
    style="background-color: rgb(105, 105, 105); color: rgb(255, 255, 255);"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
          />
        </svg>
      </span>
      Login with Google
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
